datashader/datatypes.py

```diff
@@ -0,0 +1,364 @@
+import numpy as np
+from pandas.api.extensions import (
+    ExtensionDtype, ExtensionArray, register_extension_dtype)
+from numbers import Integral
+
+
+@register_extension_dtype
+class RaggedDtype(ExtensionDtype):
```

Comment:
Does it make sense to parametrize this by the underlying `dtype` of the data? e.g. a `RaggedDtype[int]`, `RaggedDtype[float]`? Being able to tell the dtype of an output by just looking at the type, and not the values, is often helpful.

Reply:
Pandas typically calls this the `.subdtype` (see SparseArray or IntervalArray).

Reply:
Yes that makes good sense, I'll take a look at how the other extension arrays handle this.

Reply:
Done in 75f914d